### PR TITLE
Proposal. Replacing com.google.common.base.MoreObjects.firstNonNull &…

### PR DESCRIPTION
**Pull request by pks-1981**
Proposal. Replacing com.google.common.base.MoreObjects.firstNonNull & com.google.common.base.Preconditions.checkArgument

## References

https://github.com/wiremock/wiremock/issues/2111

## Submitter checklist

- [x] The PR request is well described and justified, including the body and the references
- [x] The PR title represents the desired changelog entry
- [x] The repository's code style is followed (see the contributing guide)
- [x] Test coverage that demonstrates that the change works as expected
- [ ] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

### CODE REVIEW

File: src/main/java/com/github/tomakehurst/wiremock/stubbing/AbstractScenarios.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public abstract class AbstractScenarios implements Scenarios {
 
   private final ScenariosStore store;
 
-  public AbstractScenarios(ScenariosStore store) {
+  protected AbstractScenarios(ScenariosStore store) {
```
Comment (oleg-nenashev) [inline src/main/java/com/github/tomakehurst/wiremock/stubbing/AbstractScenarios.java:30]: Revert? Seems to be a legit API

File: src/main/java/com/github/tomakehurst/wiremock/stubbing/AbstractStubMappings.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public abstract class AbstractStubMappings implements StubMappings {
   protected final List<StubLifecycleListener> stubLifecycleListeners;
   protected final StubMappingStore store;
 
-  public AbstractStubMappings(
+  protected AbstractStubMappings(
```
Comment (oleg-nenashev) [inline src/main/java/com/github/tomakehurst/wiremock/stubbing/AbstractStubMappings.java:51]: Legit API?

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Requesting changes because of API changes that do not seem to be related to the PRs purpose, or justified otherwise. Let's do them separately at least. There is also a bunch of other refactorings, but I do not mind

**Comment by pks-1981** [comment]
@oleg-nenashev 
Hello
I don't understand...
I did everything, but it's like nothing...
image

**Comment by oleg-nenashev** [comment]
@pks-1981

> I don't understand...
I did everything, but it's like nothing...

Is it still actual?

**Pull request merged**